### PR TITLE
Remove Bundler::Environment

### PR DESCRIPTION
**Pull request by segiddins**
Pretty straightforward refactoring

### CODE REVIEW

File: lib/bundler/installer.rb [hunk 1]
```diff
@@ -56,12 +61,13 @@ def self.install(root, definition, options = {})
     # Finally: TODO add documentation for how the standalone process works.
     def run(options)
       create_bundle_path
+      remove_0_9_environment_rb
```
Comment (indirect) [inline lib/bundler/installer.rb]: please, please trash this 😬

**Comment by indirect** [comment]
Pretty ridiculous that these classes only shared an `initialize` method 😛

File: lib/bundler/installer.rb (hunk 1 above)
Comment (segiddins) [inline lib/bundler/installer.rb]: there's a test for the 0.9 upgrade behavior, so to adhere to semver we should probably wait until 2.0 to do so

**Comment by segiddins** [comment]
Fix'd

**Comment by indirect** [comment]
@homu r+

**Pull request merged**